packages/@aws-cdk/aws-ec2/lib/vpc-endpoint.ts

```diff
@@ -427,16 +445,21 @@ export class InterfaceVpcEndpoint extends VpcEndpoint implements IInterfaceVpcEn
 
     const subnets = props.vpc.selectSubnets({ ...props.subnets, onePerAz: true });
     const subnetIds = subnets.subnetIds;
-
-    const endpoint = new CfnVPCEndpoint(this, 'Resource', {
-      privateDnsEnabled: props.privateDnsEnabled !== undefined ? props.privateDnsEnabled : true,
+    const endpointProps = {
+      privateDnsEnabled: true,
```

Comment:
I don't like the mutable props object you're using here. Why not simply:

```
new CfnVPCEndpoint(this, 'resource', {
  privateDnsEnabled: props.privateDnsEnabled ?? props.service.privateDnsDefault ?? true,
  // ...
});
```

Reply:
I've never seen '??' before -- did you mean a different operator?